stmhal/irq.h

```diff
@@ -39,46 +39,75 @@ MP_DECLARE_CONST_FUN_OBJ(pyb_disable_irq_obj);
 MP_DECLARE_CONST_FUN_OBJ(pyb_enable_irq_obj);
 
 // IRQ priority definitions.
+//
 // Lower number implies higher interrupt priority.
-
-#define IRQ_PRI_CAN 0x7
-#define IRQ_SUBPRI_CAN 0x0
-
-#define IRQ_PRI_DMA 0x6
-#define IRQ_SUBPRI_DMA 0x0
-
-#define IRQ_PRI_EXTINT 0xf
-#define IRQ_SUBPRI_EXTINT 0xf
+//
+// The default priority grouping is set to NVIC_PRIORITYGROUP_4 in the
+// HAL_Init function. This corresponds to 4 bits for the priority field
+// and 0 bits for the sub-priority field (which means that for all intensive
+// purposes that the sub-priorities below are ignored).
+//
+// While a given interrupt is being processed, only higher priority (lower number)
+// interrupts will preempt a given interrupt. If sub-priorities are active
+// then the sub-priority determines the order that pending interrupts of
+// a given priority are executed. This is only meaningful if 2 or more
+// interrupts of the same priority are pending at the same time.
+//
+// The priority of the SysTick timer is determined from the TICK_INT_PRIORITY
+// value which is normally set to 0 in the stm32f4xx_hal_conf.h file.
+//
+// The following interrupts are arranged from highest priority to lowest
+// priority to make it a bit easier to figure out.
+
+//                           Priority   Sub-Priority
+//                           --------   ------------
+//#def  IRQ_PRI_SYSTICK         0
+//#def  IRQ_SUBPRI_SYSTICK                  0
 
 // Flash IRQ must be higher priority than interrupts of all those components
 // that rely on the flash storage.
-#define IRQ_PRI_FLASH 0x1
-#define IRQ_SUBPRI_FLASH 0x1
+#define IRQ_PRI_FLASH           1
+#define IRQ_SUBPRI_FLASH                    1
 
-#define IRQ_PRI_OTG_FS 0x6
-#define IRQ_SUBPRI_OTG_FS 0x0
+#define IRQ_PRI_SDIO            4
```

Comment:
I think we need to add a comment here that SDIO IRQ must be higher priority than the DMA that it uses (it is, but just need a comment).